pmdamyredis.pl

```diff
@@ -925,24 +935,21 @@ sub get_redis_data {
         while ($socket->recv($cur_resp,$cfg{max_recv_len}),$cur_resp) {
             $resp .= $cur_resp;
 
-            if (not $len and not (($header,$len) = ($resp =~ /\A(\$(\d+)[\r\n]+)/))) {
+            if (defined $pass and length($pass)) {
+              # Wait for OK status reply, then payload
+              if (not $len and not (($header,$len) = ($resp =~ /\A(\+OK\r\n\$(\d+)[\r\n]+)/))) {
+                mydebug("... still do not have enough data to detect header");
+                next;
+              }
+            } else {
+              # Wait for payload
+              if (not $len and not (($header,$len) = ($resp =~ /\A(\$(\d+)[\r\n]+)/))) {
                 mydebug("... still do not have enough data to detect header");
-
                 next;
+              }
             }
 
             mydebug("Len: $len, header: '$header', response length: " . length($resp));
-
-            if ($len) {
```

Comment:
FYI, this is no longer needed as we're sending "QUIT" along with the "INFO" command so the server will close the connection as soon as it is done sending the reply.